Mathlib/Tactic/CategoryTheory/Reassoc.lean

```diff
@@ -66,29 +66,51 @@ Note that if you want both the lemma and the reassociated lemma to be
 `simp` lemmas, you should tag the lemma `@[reassoc (attr := simp)]`.
 The variant `@[simp, reassoc]` on a lemma `F` will tag `F` with `@[simp]`,
 but not `F_assoc` (this is sometimes useful).
+
+This attribute also works for lemmas of shape `∀ .., f = g` where `f g : X ≅ Y` are
+isomorphisms, provided that `Tactic.CategoryTheory.IsoReassoc` has been imported.
 -/
 syntax (name := reassoc) "reassoc" (" (" &"attr" " := " Parser.Term.attrInstance,* ")")? : attr
 
+/--
+First implementation of the `reassoc` attribute.
+Will be overriden in `Tactic.CategoryTheory.IsoReassoc.lean` to allow use on equality of
+isomorphisms.
+-/
+initialize reassocImplRef : IO.Ref (Name → Syntax → AttributeKind → AttrM Unit) ←
```

Comment:
Why not just implement it all in one big block? Do you predict that there is a lot of room for extensibility of the attribute? Is it about minimizing imports?

Reply:
It's about imports. The file `CategoryTheory.Iso` imports the current implementation of reassoc and benefits from this import, so to get it to work with isomorphisms as well it needs to be done in two phases.